### ISSUE TITLE
clean script for DHHS_CMS

### ISSUE THREAD

**Issue by rochelle-snyder** [judgelord/correspondence#153]
First and last names of members are in two separate columns labeled "From Last Name" and "From First Name." Also, some observations appear to be duplicated because they required action by multiple offices within CMS. If two or more observations appear to be identical except for "Action Office" and maybe the date columns (if the time is off by a minute or so), you can probably error out the duplicates. @judgelord I can't think of any situation in which knowing all of the action offices might be relevant for analyses (other than hand-coding), as I assume that the agency rather than the member is making this decision, but let me know if you think we should do this differently.

**Comment by rochelle-snyder** [comment]
Added other data (covering same time period) to sheet. It was in a different format, with the names in just one column (labeled "From"). It also has better dates, so the dates in the DATE column for these data are the actual correspondence dates, rather than the dates that CMS opened the correspondence.

**Comment by judgelord** [comment]
This is all right--one observation per letter. We might as well keep the action offices collapsed into a single string for each observation. I can see this as a measure of prioritization, complexity of the ask, or the subject. In general, we should try to retain all information where possible.

**Comment by rochelle-snyder** [comment]
There are a significant number of letters in the second batch of logs (starting at row 57,688 in the Google sheet) that do not have dates in the "DATE" column (which, for this second batch of logs, contains the date that was on the letter). However, these can be filled in using the dates from the "Date to CMS" column, which seems to be pretty close to the letter date.

**Comment by Maggie-Nead** [comment]
Started clean script for CMS. Fixed some name format typos and added code to draw a sample that will run through  extractMemberNames

**Comment by Maggie-Nead** [comment]
@rochelle-snyder @judgelord  There are two letters from McCollum, Bill in CMS in the 111th congress. This potentially could be McCollum, Betty who was actually in the 11th congress but there is no way for me to tell so, for now, I am marking these observations as "no longer in congress" in the ERROR column.

**Comment by Maggie-Nead** [comment]
@judgelord the names in the From column are being loaded in from the google sheet as NA, I'm not sure how to fix this as it is not because of the script.

**Comment by judgelord** [comment]
@Maggie-Nead ah ha, yes. `gs_read` uses the first 1000 entries to guess the variable type. If the first 1000 are blank, it guests NA. The fix is just to add some text in the FROM column at the top of the sheet.

However, a better fix would be to modify `gs_read` defaults to read all as strings in setup.R

**Comment by judgelord** [comment]
Duplicates are largely fixed. 

Remaining duplicates have distinct subjects:

"Regarding why Medicaid won't cover the cost of her daughter's feeding tube"
"Regarding why Medicare won't cover the cost of her daughter's feeding tube"

Yet some on the same date are not duplicates, so we can't just collapse subjects.

**Comment by judgelord** [comment]
new data cover late 2012 through the end of 2018 and look like they overlap with what we already have, but they do include the date of the actual correspondence instead of the date that they were entered into the system, so we may want to merge those into our existing data
